src/PointwiseFunctions/AnalyticData/GrMhd/KomissarovShock.cpp

```diff
@@ -0,0 +1,232 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/GrMhd/KomissarovShock.hpp"
+
+#include <cmath>
+#include <ostream>
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "Parallel/PupStlCpp11.hpp"
+#include "Utilities/ConstantExpressions.hpp"  // IWYU pragma: keep
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/Math.hpp"  // IWYU pragma: keep
+
+namespace {
+template <typename DataType>
+Scalar<DataType> compute_piecewise(
+    const tnsr::I<DataType, 3, Frame::Inertial>& x, const double& inner_value,
+    const double& outer_value) noexcept {
+  auto piecewise_scalar = make_with_value<Scalar<DataType>>(x, 0.0);
+
+  // piecewise_scalar should equal inner_value for x < 0,
+  // should equal outer_value for x > 0. Here, use step_function() to compute
+  // the result in each region separately.
+  get(piecewise_scalar) += step_function(-get<0>(x)) * inner_value;
+  get(piecewise_scalar) += step_function(get<0>(x)) * outer_value;
+  return piecewise_scalar;
+}
+template <typename DataType>
+tnsr::I<DataType, 3, Frame::Inertial> compute_piecewise_vector(
+    const tnsr::I<DataType, 3, Frame::Inertial>& x,
+    const std::array<double, 3>& inner_value,
+    const std::array<double, 3>& outer_value) noexcept {
+  auto piecewise_vector =
+      make_with_value<tnsr::I<DataType, 3, Frame::Inertial>>(x, 0.0);
+
+  // piecewise_vector should equal inner_value for x < 0,
+  // should equal outer_value for x > 0. Here, use step_function() to compute
+  // the result in each region separately.
+  get<0>(piecewise_vector) +=
```

Comment:
It looks to me that this is a shock along the x-direction only? Unless we know that we don't need a general-direction shock, it would be nice to make this a full-3D AnalyticData :smile:

Reply:
This is a 1D test, so we probably don't need general shock propagation directions. [Komissarov](https://academic.oup.com/mnras/article-abstract/303/2/343/1067242) does run another test where the propagation direction is rotated by 45deg, but not sure if we want to do that, too. It's not done in the spectre 2017 paper.